File: backend/src/main/java/com/team05/petmeeting/domain/adoption/dto/AdoptionApplyRes.java
package com.team05.petmeeting.domain.adoption.dto;

import com.team05.petmeeting.domain.adoption.entity.AdoptionStatus;
import lombok.Getter;

@Getter
public class AdoptionApplyRes {
    private Long applicationId;
    private AdoptionStatus status;
    private AnimalInfo animalInfo;

    public AdoptionApplyRes(Long applicationId, AdoptionStatus status, AnimalInfo animalInfo) {
        this.applicationId = applicationId;
        this.status = status;
        this.animalInfo = animalInfo;
    }

    @Getter
    public static class AnimalInfo {
        private String desertionNo;
        private String kindFullNm;
        private String careNm; // 보호소 이름
        private String careOwnerNm;

        public AnimalInfo(String desertionNo, String kindFullNm, String careNm, String careOwnerNm) {
            this.desertionNo = desertionNo;
            this.kindFullNm = kindFullNm;
            this.careNm = careNm;
            this.careOwnerNm = careOwnerNm;
        }
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/adoption/dto/AdoptionApplyRes.kt
package com.team05.petmeeting.domain.adoption.dto

import com.team05.petmeeting.domain.adoption.entity.AdoptionStatus

data class AdoptionApplyRes(
    val applicationId: Long,
    val status: AdoptionStatus,
    val animalInfo: AnimalInfo,
) {
    data class AnimalInfo(
        val desertionNo: String?,
        val kindFullNm: String?,
        val careNm: String?,
        val careOwnerNm: String?,
    )
}
